logger.go

```diff
@@ -5,155 +5,260 @@ import (
 	"os"
 	"runtime"
 	"strings"
+	"sync"
+	"testing"
 	"time"
 )
 
-type logWrappedT struct {
-	T
+type loggerT[ET T] struct {
+	runTHelper    // Embeds T and provides Fail/Parallel
+	orig       ET // Keep reference to original for Run method
+	logger     func(string)
+	skipFrames int // Additional skip frames for nested wrappers
+}
+
+// replaceLoggerT directly implements T interface to avoid double loggerT wrapping
```

Comment:
This is so similar to `loggerT`, I'm not sure why they both exist. I'm the sure the comment is trying to explain that, and I feel dumb but I don't get it.